macOS/debugger.cpp

```diff
@@ -239,6 +264,27 @@ void Debugger::GetLoadCommand(mach_header_64 mach_header,
   }
 }
 
+void *Debugger::MakeEntryRemoteAddress(mach_vm_address_t address, size_t size) {
+  mach_port_t shm_port;
+  if (address == 0)
+    return NULL;
+
+  memory_object_size_t memoryObjectSize = round_page(size);
+  kern_return_t ret = mach_make_memory_entry_64(mach_target->Task(), &memoryObjectSize, address, VM_PROT_READ, &shm_port, MACH_PORT_NULL);
+  if (ret != KERN_SUCCESS) {
+    FATAL("Error (%s) remote allocate share memory\n", mach_error_string(ret));
+  }
+
+  mach_vm_address_t map_address;
+  ret = mach_vm_map(mach_task_self(), &map_address, memoryObjectSize, 0, VM_FLAGS_ANYWHERE, shm_port, 0, 0, VM_PROT_READ, VM_PROT_READ, VM_INHERIT_NONE);
+  if (ret != KERN_SUCCESS) {
+    FATAL("Error (%s) map memory\n", mach_error_string(ret));
+  }
+  share_mem_port[map_address] = shm_port;
+  share_mem_map[address] = map_address;
+
+  return (void *)map_address;
+}
 
 void *Debugger::RemoteAllocateNear(uint64_t region_min,
```

Comment:
I think it might be undesirable that RemoteAllocateNear will *always* attempt to create shared memory. I think there should be an additional argument here, e.g. 

void *Debugger::RemoteAllocateNear(uint64_t region_min, uint64_t region_max, bool use_shared_memory = false);

and then you can set the third argument to 'true' in LiteCov. Note that this would also require changing the signature of RemoteAllocateNear in the Windows version (there this argument will be ignored), but should be a minor change.